### PR TITLE
Update log configurations [ECR-3139]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview
Update log configurations:
1. Configure SLF4J->Log4j binding 
2. Update log configurations:
    - For Java tests — to print to stdout our debug messages and warnings from 3rd-party libraries; 
    - For native tests — to print only fatal errors. That reduces the number of log messages in build output.
3. Redirect any output of unit and integration Java tests to files.
4. Move the log4j configuration for native ITs to native ITs
so that it is (a) easily discoverable; (b) can be changed with no
recompilation of Java `fakes` module.
5. Also, pass the path to the Log4j configuration file to the
service runtime bootstrap test to remove the error message from logs.

---
See: https://jira.bf.local/browse/ECR-3139?focusedCommentId=43246&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-43246

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/rust/integration_tests/src/vm.rs [hunk 1]
```diff
@@ -137,27 +158,30 @@ pub fn get_libpath() -> String {
...
         .unwrap()
 }
 
+/// The path to the root directory of the Rust parent module.
+fn rust_project_root_dir() -> PathBuf {
+    project_root_dir().join("..").canonicalize().unwrap()
+}
+
+/// The path to `integration_tests` root directory.
+fn project_root_dir() -> &'static Path {
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/rust/integration_tests/src/vm.rs]: Does it have to return a reference? If so when is the object destroyed?

File: exonum-java-binding/CHANGELOG.md [hunk 2]
```diff
@@ -19,7 +19,9 @@ The release is based on Exonum 0.11
 
 ### Added
 - `toOptional()` method to `EntryIndexProxy`. (#790)
-- `getPoolTransactionHashes()` method to `Blockchain`. (#850)
+- `getTransactionPool()` method to `Blockchain`. (#850)
+- SLF4J to Log4j binding to enable libraries coded to the SLF4J API to use Log4j 2, 
+  used by Exonum Java, as the implementation.
```
Comment (bullet-tooth) [inline exonum-java-binding/CHANGELOG.md]:
```suggestion
  used by Exonum Java, as the implementation. (#854)
```

File: exonum-java-binding/CHANGELOG.md [hunk 3]
```diff
@@ -19,7 +19,9 @@ The release is based on Exonum 0.11
 
 ### Added
 - `toOptional()` method to `EntryIndexProxy`. (#790)
-- `getPoolTransactionHashes()` method to `Blockchain`. (#850)
+- `getTransactionPool()` method to `Blockchain`. (#850)
+- SLF4J to Log4j binding to enable libraries coded to the SLF4J API to use Log4j 2,
```
Comment (bullet-tooth) [inline exonum-java-binding/CHANGELOG.md]: I believe that "- SLF4J support" is enough in the changelog. Also, I would mention in docs that EJB provides `log4j 2` as a logging implementation

File: exonum-java-binding/pom.xml [hunk 4]
```diff
@@ -354,6 +354,7 @@
           <artifactId>maven-surefire-plugin</artifactId>
           <version>2.22.1</version>
           <configuration>
+            <redirectTestOutputToFile>true</redirectTestOutputToFile>
```
Comment (bullet-tooth) [inline exonum-java-binding/pom.xml]: What does it bring to us? We do not use stdout in tests so often.

Comment (dmitry-timofeev) [reply exonum-java-binding/pom.xml]: We don't, the logger prints to stdout (see the docs)

File: exonum-java-binding/core/rust/integration_tests/src/vm.rs [hunk 5]
```diff
@@ -103,26 +105,45 @@ fn get_fakes_classpath_option() -> String {
...
     // should be used `;` as path separator on Windows [https://jira.bf.local/browse/ECR-587]
     format!("{}:{}", class_path, fakes_classes)
 }
 
+/// Returns a Log4j system property pointing to the configuration file. The file is in
+/// the base directory of the `integration_tests` project and must be edited if more detailed
+/// Java logs are needed for debugging purposes.
+///
+/// It requires the log4j-core library to be present on the classpath, which is the case with fakes.
+fn get_log4j_path_option() -> String {
```
Comment (vitvakatu) [inline exonum-java-binding/core/rust/integration_tests/src/vm.rs]: nit: why we even use these `get_` prefixes? https://rust-lang-nursery.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

File: exonum-java-binding/core/rust/integration_tests/src/vm.rs (hunk 1 above)
Comment (vitvakatu) [reply exonum-java-binding/core/rust/integration_tests/src/vm.rs]: I guess we need to return `PathBuf`, not `&Path`.

**Pull request merged**